### PR TITLE
Make affinity configurable

### PR DESCRIPTION
**Pull request by jsanda**
Fixes https://github.com/k8ssandra/k8ssandra/issues/847.

Note that when this is merged we also need to update the reaper-operator chart in k8ssandra including the Reaper CRD.

### CODE REVIEW

**Review by jdonenine — COMMENTED** [review]
Was able to run a few different tests with the test configurations you provided, thanks for adding those, made things much easier.  All of the testing looked good.  I just had one question about the generated files, but assuming that my understanding is correct, I think we're good to go.

File: api/v1alpha1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -145,6 +145,11 @@ func (in *ReaperList) DeepCopyObject() runtime.Object {
 func (in *ReaperSpec) DeepCopyInto(out *ReaperSpec) {
 	*out = *in
 	in.ServerConfig.DeepCopyInto(&out.ServerConfig)
+	if in.Affinity != nil {
```
[the remark this answers is not in the document]
Comment (jsanda) [reply api/v1alpha1/zz_generated.deepcopy.go:148]: Yes, the zz_generate.deepcopy.go file is auto generated. It should not be modified by hand. 

And yes the CRD changes are auto generated as well. It gets generated based on the contents of https://github.com/k8ssandra/reaper-operator/blob/master/api/v1alpha1/reaper_types.go.

[the remark this answers is not in the document]
Comment (jdonenine) [reply api/v1alpha1/zz_generated.deepcopy.go:148]: Cool, thanks!

**Review by jdonenine — APPROVED** [review]
👍

**Pull request merged**